src/lib/validateLocalDocument.ts

```diff
@@ -47,3 +48,26 @@ export default async function validateLocalDocument(
 
   return (await Promise.all(validationPromises)).flat();
 }
+
+/**
+ * Validates jsonDocument and returns all validation results where the result's
+ * affected fields contain the `validationFieldName`
+ * @param jsonDocument
+ * @param validationFieldName The name of the field name in camelCase notation.
+ * @param rules the rules to validate against
+ * @returns an array of ValidationResult
+ */
+export async function validateField(
+  jsonDocument: string | object,
+  validationFieldName: string,
+  rules: ValidationRule[]
+) {
+  const results = validateLocalDocument(jsonDocument, rules);
```

Comment:
Better to lift the await up to here, instead of doing it in the next line, as you're not actually wanting to fork the promise chain.